clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5209,6 +5220,92 @@ static void handleIntelFPGAMergeAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
                  IntelFPGAMergeAttr(S.Context, AL, Results[0], Results[1]));
 }
 
+/// Handle the bank_bits attribute.
+/// This attribute accepts a list of values greater than zero.
+/// This is incompatible with the register attribute.
+/// The numbanks and bank_bits attributes are related. If numbanks exists
+/// when handling bank_bits they are checked for consistency. If numbanks
+/// hasn't been added yet an implicit one is added with the correct value.
```

Comment:
Why is that needed? I'm probably just missing something in all this memory attributes stuff...

Reply:
Because bank_bits and numbanks are related attributes - by using the bank_bits attribute, you can separate memory data into multiple banks while specifying which address bits to use to select the bank.